src/release.js

```diff
@@ -100,22 +100,41 @@ module.exports = async function ({ github, context, inputs }) {
     const opticToken = inputs['optic-token']
     const npmToken = inputs['npm-token']
     const provenance = /true/i.test(inputs['provenance'])
+    const access = inputs['access']
+
+    // Can't limit custom action inputs to fixed options like "choice" type in a manual action
+    const validAccessOptions = ['public', 'restricted']
+    if (access && !validAccessOptions.includes(access)) {
+      core.setFailed(
+        `Invalid "access" option provided ("${access}"), should be one of "${validAccessOptions.join(
+          '", "'
+        )}"`
+      )
+      return
+    }
+
+    let publishOptions = {
+      npmToken,
+      opticToken,
+      opticUrl,
+      npmTag,
+      version,
+      provenance,
+      access,
+    }
 
-    // Fail fast with meaningful error if user wants provenance but their setup won't deliver
     if (provenance) {
+      // Fail fast with meaningful error if user wants provenance but their setup won't deliver,
+      // and apply any necessary options tweaks.
       const npmVersion = await getNpmVersion()
-      checkProvenanceViability(npmVersion)
+      publishOptions = await ensureProvenanceViability(
+        npmVersion,
+        publishOptions
+      )
```

Comment:
can we mutate the options provided to this function instead of returning them and having to overwrite the publishOptions variable?